Source/Core/VideoCommon/TextureCacheBase.cpp

```diff
@@ -798,7 +812,18 @@ TextureCacheBase::TCacheEntryBase* TextureCacheBase::Load(const u32 stage)
   entry->is_custom_tex = hires_tex != nullptr;
 
   // load texture
-  entry->Load(temp, width, height, expandedWidth, 0);
+  if (decode_on_gpu)
+  {
+    const u8* tlut = &texMem[tlutaddr];
+    u32 row_stride = bytes_per_block * (expandedWidth / bsw);
+    g_texture_cache->DecodeTextureOnGPU(
```

Comment:
What do you think about s/DecodeTextureOnGPU/DecodeTexture/. Always call this one here, and the backend (and so the default implementation) shall just call the texture decoder + entry->Load on failure.